gcloud/search/document.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Document."""
+
+import datetime
+
+import six
+
+from gcloud._helpers import UTC
+from gcloud._helpers import _RFC3339_MICROS
+from gcloud.exceptions import NotFound
+
+
+class StringValue(object):
```

Comment:
RE: 

> Could be, I don't see an win for a named tuple, really.

----

The win is the reduction of boilerplate. These data classes have no behavior, they just hold data.

----

We've now got lines 26-50 for this class and could do it with:

```python
StringValue = namedtuple('StringValue',
                         ['string_value', 'string_format', 'language'])
StringValue.value_type = 'string'
StringValue.__new__.__defaults__ = (None, None)
```